server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -324,6 +324,8 @@ private static void ensureSnapshotNameNotRunning(List<SnapshotsInProgress.Entry>
                 .collect(Collectors.toMap(IndexId::getName, Function.identity()));
     }
 
+    // TODO: It is worth revisiting the design choice of creating a placeholder entry in snapshots-in-progress here once we have a cache
```

Comment:
Just a thought I had while writing this up. The placeholder logic might be questionable but we can easily remove it in a BwC manner like we did for snapshot create if want to in a follow-up so I'm not too worried here either.